src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -658,8 +679,29 @@ protected void Accepting()
             Receive<EndpointWriter.TookOver>(tookover => RemovePendingReader(tookover.Writer, tookover.ProtocolHandle));
             Receive<ReliableDeliverySupervisor.GotUid>(gotuid =>
             {
-                _endpoints.RegisterWritableEndpointUid(gotuid.RemoteAddress, gotuid.Uid);
-                HandleStashedInbound(Sender, writerIsIdle: false);
+                
+                var policy = _endpoints.WritableEndpointWithPolicyFor(gotuid.RemoteAddress);
```

Comment:
During a `GotUid` message we check to see if the endpoint we've just received a UID for has a quarantined policy or not. If it does, we avoid overwriting the quarantine and stay in a quarantined state. Otherwise we register a writeable endpoint as usual.